waku/v2/protocol/waku_store/protocol.nim

```diff
@@ -213,19 +225,26 @@ proc handleMessage*(w: WakuStore, pubsubTopic: string, msg: WakuMessage) =
     # The message is ephemeral, should not be stored
     return
   
+  if not isValidMessage(msg):
+    waku_store_errors.inc(labelValues = [invalidMessage])
+    return
+
+
   let insertStartTime = getTime().toUnixFloat()
-    
-  let now = getNanosecondTime(getTime().toUnixFloat())
-  let index = Index.compute(msg, receivedTime=now, pubsubTopic=pubsubTopic)
+  
+  block:
+    let
+      msgDigest = computeDigest(msg) 
+      msgReceivedTime = if msg.timestamp > 0: msg.timestamp
+                        else: getNanosecondTime(getTime().toUnixFloat()) 
```

Comment:
~Not sure of real impact here, but I think that getting current epoch time is a relatively expensive operation. Should we consider doing this once at the start of `handleMessage` and use it both for the validity check and received time?~

On second thought, probably not that important as this second `getNanosecondTime()` will only be called in the presumably rare case when `msg.timestamp == 0`